src/main/kotlin/sliv/tool/scene/view/SceneView.kt

```diff
@@ -14,27 +15,33 @@ class SceneView : View() {
         }
     }
 
-    //TODO: some virtualized pane to place frames
     override val root = vbox {
         label("Empty scene placeholder")
     }
 
     private fun draw() {
         root.children.clear()
-        root.add(vbox {
-            val scene = controller.scene.value
-            val frames = (0 until scene.framesCount).map { i -> scene.getFrame.invoke(i)!! }
-            scrollpane {
-                gridpane {
-                    frames.forEachIndexed { index, frame ->
-                        group {
-                            add(FrameView(600.0, 600.0, frame))
-                        }.gridpaneConstraints {
-                            columnRowIndex(0, index)
-                        }
-                    }
-                }
+
+        val scene = controller.scene.value
+        val frames = (0 until scene.framesCount).map { i -> scene.getFrame.invoke(i)!! }
+
+        //TODO: all cells should have the same size. I can get it from the image of the first frame. What to do if some images have another size (irregular case).
+        val width = 500.0
+        val height = 500.0
+
+        val margin = 10.0
+
+        val columnsNumber = 30 //TODO: should be set from the UI
+
+        val grid =
+            VirtualizedFXGridProvider.createGrid(frames, columnsNumber, width + margin, height + margin) { frame ->
```

Comment:
`val grid = VirtualizedFXGridProvider.createGrid(` on first line and then argument on each new line